main.cpp

```diff
@@ -307,18 +218,23 @@ struct DoubleType
     DoubleType &subtract(double rhs);
     DoubleType &multiply(double rhs);
     DoubleType &divide(double rhs);
+    DoubleType &pow(double exp);
+    DoubleType &pow(const FloatType &exp);
+    DoubleType &pow(const DoubleType &exp);
+    DoubleType &pow(const IntType &exp);
```

Comment:
Don't put `&` next to the function name. 
put it next to the return type.
It looks like you're taking the address of the function, instead of returning a reference.
- [ ] revise